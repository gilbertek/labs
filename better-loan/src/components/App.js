import React, { Component } from 'react';
import PropTypes from 'prop-types';

class App extends Component {
  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string
  };

  render() {
    return (
      <div className="wrapper">
        <p>Hello World!</p>
      </div>
    );
  }
}

export default App;
