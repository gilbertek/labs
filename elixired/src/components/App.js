import React, { Component } from 'React';
import ClinicalMedication from './ClinicalMedication';
import ClinicalMedicationForm from './ClinicalMedicationForm';
import MedicationHistory from './MedicationHistory';
import Prescription from './Prescription';
import MedicationHistoryReview from './MedicationHistoryReview';

class App extends Component {
  render() {
    const props = { memberId: 1, editMode: false };
    const editprops = { memberId: 1, editMode: true };

    return (
      <div className='wrapper'>
        <ClinicalMedication {...props} />
        <br />
        <br />
        <ClinicalMedicationForm />
        <br />
        <br />
        <MedicationHistory {...props} />
        {/*<MedicationHistory {...editprops} />*/}
        <br />
        <br />
        <MedicationHistoryReview {...editprops} />
        <br />
        <br />
        <Prescription />
      </div>
    );
  }
}

export default App;
