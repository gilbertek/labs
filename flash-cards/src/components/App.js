import React from 'React';
import Sidebar from './Sidebar/Sidebar';

const App = () =>
  <div className='container'>
    <div className='row'>
      <Sidebar decks={[ {name: 'Deck 1'} ]} addingDeck={false} />
    </div>
  </div>

export default App;
