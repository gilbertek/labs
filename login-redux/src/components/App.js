import React from 'React';

const App = () =>
  <div className='container'>
    <div className='row'>
      <p>Hello, World!</p>
    </div>
  </div>

export default App;
