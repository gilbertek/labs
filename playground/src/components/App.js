import React, { Component } from "React";
import User from './User';
import Phone from './Phone';

class App extends Component {
  render() {
    return (
      <div className='wrapper'>
        <User />
        <br />
        <br />
        <br />
        <Phone />
      </div>
    );
  }
}

export default App;
