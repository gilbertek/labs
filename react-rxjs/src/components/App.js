import React, { Component } from 'React';

class App extends Component {
  render() {
    return (
      <p>Hello Rxjs</p>
    );
  }
}

export default App;
