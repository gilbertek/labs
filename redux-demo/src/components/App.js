import React, { Component } from 'React';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import propTypes from 'prop-types';
import HomePage from './home/HomePage';
import AboutPage from './about/AboutPage';

class App extends Component {
  render() {
    return(
      <div className='container-fluid'>
        <div className='row'>
          <Router>
            <Switch>
              <Route exact path='/' component={HomePage} />

              <Route path='/about' component={AboutPage} />

              <Route render={function () {
                return <p>Not Found...</p>
              }} />
            </Switch>
          </Router>
        </div>
      </div>
    );
  }
}

App.propTypes = {
  children: propTypes.object
};
export default App;
