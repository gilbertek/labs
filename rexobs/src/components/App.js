import React, { Component } from 'React';
import Stories from './Stories';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Stories />
      </div>
    );
  }
}

export default App;
