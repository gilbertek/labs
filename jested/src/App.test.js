import React from 'react';
import ReactDOM from 'react-dom';
import App, { Link } from './App';
import { configure, shallow, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

configure({ adapter: new Adapter() })

describe('<App /> shallow rendering', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('matches the snapshot', () => {
    const tree = shallow(<App />);
    expect(toJson(tree)).toMatchSnapshot();
  });


  it('should contain 1 p element', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('p.App-intro').exists()).toBe(true)
  });

  it('on button click changes p text', () => {
    const wrapper = shallow(<App />);
    const button = wrapper.find('button');
    expect(wrapper.find('.button-state').text()).toBe('No!');

    button.simulate('click');
    expect(wrapper.find('.button-state').text()).toBe('Yes!');
  });

  it('on input change, title change text', () => {
    const wrapper = shallow(<App />);
    const input = wrapper.find('input');
    expect(wrapper.find('h2').text()).toBe('');

    input.simulate('change', {currentTarget: { value: 'Georgy' }})
    expect(wrapper.find('h2').text()).toBe('Georgy');
  });

  it('updates component className with new state', () => {
    const wrapper = shallow(<App />);
    expect(wrapper.find('.blue').length).toBe(1);
    expect(wrapper.find('.red').length).toBe(0);
    wrapper.setState({ mainColor: 'red' });
    expect(wrapper.find('.blue').length).toBe(0);
    expect(wrapper.find('.red').length).toBe(1);
  });

  it('calls componentDidMount, update p tag', () => {
    jest.spyOn(App.prototype, 'componentDidMount');
    const wrapper = shallow(<App />);
    expect(App.prototype.componentDidMount.mock.calls.length).toBe(1);
    expect(wrapper.find('.lifeCycle').text()).toBe('componentDidMount');
  });

  it('setProps calls componentWillReceiveProps', () => {
    jest.spyOn(App.prototype, 'componentWillReceiveProps');
    const wrapper = shallow(<App />);
    wrapper.setProps({ hide: true });
    expect(App.prototype.componentWillReceiveProps.mock.calls.length).toBe(1);
    expect(wrapper.find('.lifeCycle').text()).toBe('componentWillReceiveProps');
  });

  it('handleStrings function returns correctly', () => {
    const wrapper = shallow(<App />);
    const trueReturn = wrapper.instance().handleStrings('Hello World');
    expect(trueReturn).toBe(true);

    const falseReturn = wrapper.instance().handleStrings('');
    expect(falseReturn).toBe(false);
  });


});

describe('<App /> mount rendering', () => {
  it('renders without crashing', () => {
    const wrapper = mount(<App />);
    expect(wrapper.find('h1').text()).toBe('Welcome to React')
  });
});

describe('<Link />', () => {
  it('link component accepts address props', () => {
    const wrapper = shallow(<Link address='www.google.com' />);
    expect(wrapper.instance().props.address).toBe('www.google.com')
  });

  it('a tag node renders href correctly', () => {
    const wrapper = shallow(<Link address='www.google.com' />);
    expect(wrapper.props().href).toBe('www.google.com');
  });

  it('returns null with tru hide props', () => {
    const wrapper = shallow(<Link hide={false} />)
    expect(wrapper.find('a').length).toBe(1)

    wrapper.setProps({ hide: true });
    expect(wrapper.get(0)).toBeNull()
  });

});

