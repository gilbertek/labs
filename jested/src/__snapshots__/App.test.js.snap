// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> shallow rendering matches the snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <h1
      className="App-title"
    >
      Welcome to React
    </h1>
    <h3
      className="blue"
    >
      Everyone is welcome
    </h3>
  </header>
  <Connect(TodoList) />
  <Form />
  <p
    className="App-intro"
  >
    To get started, edit 
    <code>
      src/App.js
    </code>
     and save to reload.
  </p>
  <p
    className="button-state"
  >
    No!
  </p>
  <button
    onClick={[Function]}
  >
    Click
  </button>
  <h2 />
  <input
    onChange={[Function]}
  />
  <p
    className="lifeCycle"
  >
    componentDidMount
  </p>
</div>
`;
